apps/factory/src/components/copy_button.tsx

```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import IconButton from "@mui/material/IconButton";
+import FileCopyIcon from "@mui/icons-material/FileCopy";
+import toast from "react-hot-toast";
+
+interface CopyToClipboardButtonProps {
+  textToCopy: string;
+}
+
+const CopyToClipboardButton: React.FC<CopyToClipboardButtonProps> = ({
+  textToCopy,
+}) => {
+  const handleCopy = () => {
+    navigator.clipboard.writeText(textToCopy);
+    toast.success("Copied");
+  };
+
+  return (
+    <>
+      <IconButton
+        onClick={handleCopy}
+        sx={{
+          margin: 1,
+        }}
+      >
+        <FileCopyIcon />
+      </IconButton>
+    </>
+  );
+};
+
+export default CopyToClipboardButton;
```

Comment:
@LovedeepSingh19 use this component other plces as well, there are 2 places where we are copy options. Use this component there as well

- [ ] One in info popup
- [ ] Share Button in the top navigation